src/classes/GRER_SyncGradeLevel_TDTM.cls

```diff
@@ -58,22 +61,53 @@ global with sharing class GRER_SyncGradeLevel_TDTM extends hed.TDTM_Runnable {
         //Declare and initiate lists to update
         List<Contact> contactsToUpdate = new List<Contact>();
         List<Grade_Enrollment__c> gradeEnrolsToUpdate = new List<Grade_Enrollment__c>();
+        
+        //Declare and initiate oldMap
+        Map<ID, Grade_Enrollment__c> oldMap = new Map<ID, Grade_Enrollment__c>();
+        if (listOld != NULL) {
```

Comment:
Should we move this to the bottom and build the map right before it is used so that it does not need to be built all the time? Also, let's make sure the top section is only for variable declaration.

Reply:
Hmm we are using the map as soon as we enter //After INSERT and After UPDATE block, so I put in at the top. Also, listOld != NULL helps? Please let me know.

Reply:
Right, we can build the map the first thing in AfterInsert/AfterUpdate block. We can still do the listOld != NULL check.